core/domain/html_validation_service.py

```diff
@@ -94,7 +94,7 @@ def wrap_with_siblings(tag: bs4.element.Tag, p: bs4.element.Tag) -> None:
         p: bs4.element.Tag. The new p tag in soup in which the tag and
             its siblings are to be wrapped.
     """
-    independent_parents = ['p', 'pre', 'ol', 'ul', 'blockquote']
+    independent_parents = ['h1', 'p', 'pre', 'ol', 'ul', 'blockquote']
```

Comment:
Will this show up in all RTEs or just the blog one?

Reply:
'h1' tag will only be present in html string if Headers are enabled in the RTE. (Currently, headers are enabled in blog editors only).